bokeh/io.py

```diff
@@ -520,6 +534,57 @@ def push(session_id=None, url=None, app_path=None, document=None, state=None, io
 
 def push_notebook(document=None, state=None, handle=None):
     ''' Update Bokeh plots in a Jupyter notebook output cells with new data
+        or property values.
+
+        When working the the notebook, the ``show`` function can be passed the
+        argument ``notebook_handle=True``, which will cause it to return a
+        handle object that can be used to update the Bokeh output later. When
+        ``push_notebook`` is called, any property updates (e.g. plot titles or
+        data source values, etc.) since the last call to ``push_notebook`` or
+        the original ``show`` call are applied to the Bokeh output in the
+        previously rendered Jupyter output cell.
+
+        Several example notebooks can be found in the GitHub repository in
+        the :bokeh-tree:`examples/howto/notebook_comms` directory.
+
+        Args:
+
+            document (Document, optional) :
+                A :class:`~bokeh.document.Document` to push from. If None,
+                uses ``curdoc()``.
+
+            state (State, optional) :
+                A Bokeh State object
+
+        Returns:
+            None
+
+        Examples:
+
+            Typical usage is typically similar to this:
+
+            .. code-block:: python
+
+                from bokeh.plotting import figure
+                from bokeh.io import output_notebook, push_notebook, show
+
+                output_notebook()
+
+                plot = figure()
+                plot.circle([1,2,3], [4,6,5])
+
+                handle = show(plot, notebook_handle=True)
+
+                # Update the plot title in the earlier cell
+                plot.title = "New Title"
+                push_notebook(handle=handle)
+
+    '''
+    deprecated((0, 12, 5), "push_notebook", "push_jupyter")
+    push_jupyter(document, state, handle)
```

Comment:
Here's the exception that proves the rule. I think `push_notebook` is too well-established to deprecate. But also, since presumably only one notebook can be "activated" at a time, and this is only useful afterwards it seems ok to leave it `push_notebook`